### PR TITLE
fix: change everything to append to the body by default

### PR DESCRIPTION
**Pull request by cal-smith**
Also add `appendInline` for cases where the component need to be within the normal page flow.

### CODE REVIEW

File: src/dialog/dialog.directive.ts [hunk 1]
```diff
@@ -36,61 +36,55 @@ export class DialogDirective implements OnInit, OnDestroy, OnChanges {
...
 	@Input() gap = 0;
 	/**
+	 * Deprecated. Defaults to true. Use appendInline to keep dialogs within page flow
 	 * Value `true` sets Dialog be appened to the body (to break out of containers)
-	 * @type {boolean}
-	 * @memberof DialogDirective
 	 */
-	@Input() appendToBody = false;
+	@Input() set appendToBody(v: boolean) {
+		console.log("appendToBody has been deprecated. dialogs now append to the body by default.");
```
Comment (zvonimirfras) [inline src/dialog/dialog.directive.ts]:
```suggestion
		console.log("`appendToBody` has been deprecated. Dialogs now append to the body by default.");
```

File: src/dialog/dialog.directive.ts [hunk 2]
```diff
@@ -36,61 +36,55 @@ export class DialogDirective implements OnInit, OnDestroy, OnChanges {
...
+		console.log("ensure you have an ibm-placeholder in your app.");
```
Comment (zvonimirfras) [inline src/dialog/dialog.directive.ts]:
```suggestion
		console.log("Ensure you have an `ibm-placeholder` in your app.");
```

File: src/dialog/dialog.directive.ts [hunk 3]
```diff
@@ -36,61 +36,55 @@ export class DialogDirective implements OnInit, OnDestroy, OnChanges {
...
+		console.log("use appendInline if you need to position your dialogs within the normal page flow");
```
Comment (zvonimirfras) [inline src/dialog/dialog.directive.ts]:
```suggestion
		console.log("Use `appendInline` if you need to position your dialogs within the normal page flow.");
```

File: src/dialog/dialog.service.ts [hunk 4]
```diff
@@ -103,13 +103,16 @@ export class DialogService {
...
 		if (!this.dialogRef) {
-			// holder for either the provided view, or the view from Placeholder
-			if (dialogConfig.appendToBody) {
-				// add our component to the placeholder
-				this.dialogRef = this.placeholderService.createComponent(this.componentFactory, this.injector);
-			} else {
+			if (dialogConfig.appendInline) {
 				// add our component to the view
 				this.dialogRef = viewContainer.createComponent(this.componentFactory, 0, this.injector);
+			} else if (this.placeholderService.hasPlaceholderRef()) {
```
Comment (zvonimirfras) [inline src/dialog/dialog.service.ts]: Did you mean
```suggestion
			} else if (!this.placeholderService.hasPlaceholderRef()) {
```

File: src/dialog/dialog.service.ts [hunk 5]
```diff
@@ -140,10 +143,10 @@ export class DialogService {
...
 		if (this.dialogRef) {
 			let elementToFocus = this.dialogRef.instance.dialogConfig["previouslyFocusedElement"];
-			if (this.dialogRef.instance.dialogConfig.appendToBody) {
-				this.placeholderService.destroyComponent(this.dialogRef);
-			} else {
+			if (this.dialogRef.instance.dialogConfig.appendInline) {
 				viewContainer.remove(viewContainer.indexOf(this.dialogRef.hostView));
+			} else if (this.placeholderService.hasPlaceholderRef()) {
+				this.placeholderService.destroyComponent(this.dialogRef);
 			}
```
Comment (zvonimirfras) [inline src/dialog/dialog.service.ts]: Do we need an `else` for the ones that are actually in the `body`?

File: src/dropdown/dropdown.component.ts [hunk 6]
```diff
@@ -74,45 +73,50 @@ import { I18n } from "./../i18n/i18n.module";
...
 	@Input() disabled = false;
 	/**
+	 * Deprecated. Dropdown now defaults to appending inline
 	 * Set to `true` if the `Dropdown` is to be appended to the DOM body.
-	 * @type {boolean}
-	 * @memberof Dropdown
 	 */
-	@Input() appendToBody = false;
+	@Input() set appendToBody (v) {
+		console.log("appendToBody has been deprecated. dropdowns now append to the body by default.");
```
Comment (zvonimirfras) [inline src/dropdown/dropdown.component.ts]:
```suggestion
		console.log("`appendToBody` has been deprecated. Dropdowns now append to the body by default.");
```

File: src/dropdown/dropdown.component.ts [hunk 7]
```diff
@@ -74,45 +73,50 @@ import { I18n } from "./../i18n/i18n.module";
...
+		console.log("ensure you have an ibm-placeholder in your app.");
```
Comment (zvonimirfras) [inline src/dropdown/dropdown.component.ts]:
```suggestion
		console.log("Ensure you have an `ibm-placeholder` in your app.");
```

File: src/dropdown/dropdown.component.ts [hunk 8]
```diff
@@ -74,45 +73,50 @@ import { I18n } from "./../i18n/i18n.module";
...
+		console.log("use appendInline if you need to position your dropdowns within the normal page flow");
```
Comment (zvonimirfras) [inline src/dropdown/dropdown.component.ts]:
```suggestion
		console.log("Use `appendInline` if you need to position your dropdowns within the normal page flow.");
```

File: src/modal/modal.component.ts [hunk 9]
```diff
@@ -44,7 +44,7 @@ import { cycleTabs } from "./../common/tab.service";
  * 			<ibm-modal-header (closeSelect)="closeModal()">Header text</ibm-modal-header>
  * 			<section class="modal-body">
  * 			<h1>Sample modal works.</h1>
- * 			<button class="btn--icon-link" nPopover="Hello there" title="Popover title" placement="right" appendToBody="false">
+ * 			<button class="btn--icon-link" nPopover="Hello there" title="Popover title" placement="right">
```
Comment (zvonimirfras) [inline src/modal/modal.component.ts]: If `appendToBody` was set to `false`, wouldn't we want to set `appendInline` to `true`?
```suggestion
 * 			<button class="btn--icon-link" nPopover="Hello there" title="Popover title" placement="right" appendInline="true">
```

File: src/dialog/dialog.service.ts (hunk 5 above)
Comment (cal-smith) [reply src/dialog/dialog.service.ts]: it didn't have one before, so I guess not?

File: src/dialog/tooltip/tooltip.directive.spec.ts [hunk 10]
```diff
@@ -39,7 +39,7 @@ describe("Tooltip directive", () => {
 	it("should create the tooltip component and tooltip should appear at the top", () => {
 		TestBed.overrideComponent(TooltipTest, {
 			set: {
-				template: "<button ibmTooltip='Hello There' placement='top'>Me</button>"
+				template: "<button ibmTooltip='Hello There' placement='top'>Me</button><ibm-placeholder></ibm-placeholder>"
```
Comment (zvonimirfras) [inline src/dialog/tooltip/tooltip.directive.spec.ts]: Let's make this prettier.
```suggestion
				template: `
					<button ibmTooltip="Hello There" placement="top">Me</button>
					<ibm-placeholder></ibm-placeholder>
				`
```

**Pull request merged**